File: roman-numerals/py/roman.py
class RomanNumeral():

    def convert(self, arabic):
        units    = [1000, 900, 500, 400,  100,  90,   50,   40,  10,    9,  5,    4,    1]
        numerals = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I']
        roman = ''

        for size, numeral in zip(units, numerals):
            while arabic >= size:
                roman += numeral
                arabic -= size

        return roman

File: roman-numerals/cs/Roman.cs
using System;
using System.Collections.Generic;

public class Roman
{
    Dictionary<string, int> numerals;

    public Roman ()
    {
        numerals = new Dictionary<string, int>();
        numerals.Add("I", 1);
        numerals.Add("II", 2);
        numerals.Add("III", 3);
        numerals.Add("IV", 4);
    }

    public int Convert(string numeral)
    {
        return numerals[numeral];
    }
}